### ISSUE TITLE
concurrent_schema_changes_test.py:TestConcurrentSchemaChanges.create_lots_of_tables_concurrently_test fails with smp=2

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#1009]
running concurrent_schema_changes_test.py:TestConcurrentSchemaChanges.create_lots_of_tables_concurrently_test passes with smp=1

running it with smp=2 failes

to run it with smp=2 use `export SCYLLA_EXT_OPTS="--smp 2"`

**Comment by penberg** [comment]
@pdziepak @slivne The test times out in the debug build even with ``--smp 1``. Did anyone bisect this?

**Comment by tgrabiec** [comment]
@penberg That dtest was only added few days ago, it could be that it never passed.

**Comment by slivne** [comment]
It does pass on smp=1

**Comment by tgrabiec** [comment]
It still could never pass with smp=2, so bisecting it may be futile.

**Comment by slivne** [comment]
You are correct it never AFAIK it never passed with smp=2.

**Comment by penberg** [comment]
The bug report does not say _how_ the test fails. I am seeing ``bad_alloc`` with ``--smp 2``:

```
create_lots_of_tables_concurrently_test (concurrent_schema_changes_test.TestConcurrentSchemaChanges) ... ERROR

======================================================================
ERROR: create_lots_of_tables_concurrently_test (concurrent_schema_changes_test.TestConcurrentSchemaChanges)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/penberg/scylla-dtest/concurrent_schema_changes_test.py", line 146, in create_lots_of_tables_concurrently_test
    results = execute_concurrent(session, cmds, raise_on_first_error=True, concurrency=200)
  File "cassandra/concurrent.py", line 88, in cassandra.concurrent.execute_concurrent (cassandra/concurrent.c:1404)
    return executor.execute(concurrency, raise_on_first_error)
  File "cassandra/concurrent.py", line 185, in cassandra.concurrent.ConcurrentExecutorListResults.execute (cassandra/concurrent.c:4718)
    return super(ConcurrentExecutorListResults, self).execute(concurrency, fail_fast)
  File "cassandra/concurrent.py", line 111, in cassandra.concurrent._ConcurrentExecutor.execute (cassandra/concurrent.c:2059)
    return self._results()
  File "cassandra/concurrent.py", line 199, in cassandra.concurrent.ConcurrentExecutorListResults._results (cassandra/concurrent.c:5459)
    with self._condition:
  File "cassandra/concurrent.py", line 203, in cassandra.concurrent.ConcurrentExecutorListResults._results (cassandra/concurrent.c:5397)
    self._raise(self._exception)
  File "cassandra/concurrent.py", line 151, in cassandra.concurrent._ConcurrentExecutor._raise (cassandra/concurrent.c:3422)
    raise exc
ServerError: <ErrorMessage code=0000 [Server error] message="std::bad_alloc">
-------------------- >> begin captured logging << --------------------
dtest: DEBUG: cluster ccm directory: /home/penberg/.dtest/dtest-NLXy1v
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('606ab166-5693-39bf-8c1c-9a0b99282ce3'): ['127.0.0.1'], UUID('9e448481-5822-3e44-a4ee-eaada4533700'): ['127.0.0.2'], UUID('fbe93f66-b9ca-3753-bcf3-51637c76e14a'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('13ddb86a-3457-345b-8b66-071ac04189e7'): ['127.0.0.2'], UUID('acbe7ec3-020e-343f-9aab-72a3dd9fb804'): ['127.0.0.1'], UUID('1630c5d1-b575-357b-8536-21e78c81dddb'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('79729711-9c1d-3331-ac73-863ef762c867'): ['127.0.0.1'], UUID('d59ad738-e9d7-3039-ba29-60cc56684d23'): ['127.0.0.3'], UUID('d70d4d8e-d337-3395-9b3a-384ab494549b'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('419dea02-6cb5-3110-bba3-4533eb30e700'): ['127.0.0.1'], UUID('7a5aeae1-60cd-3929-a750-a032dc3e2913'): ['127.0.0.2'], UUID('4dce21da-21d1-389b-bacd-2b5e19983eca'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('c31fcfe4-12bf-3d2d-8e89-2ef653b96735'): ['127.0.0.3'], UUID('9f1480ab-acbd-3f80-96b9-3931c1ce5dac'): ['127.0.0.1'], UUID('37840568-3cc7-3215-97a1-7a9f8a00b7d8'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('1f94e962-2121-3bb8-848f-8da4272e7851'): ['127.0.0.1'], UUID('b4e704e3-8154-3db7-8d64-9f5f2c365144'): ['127.0.0.3'], UUID('e920ea46-6479-301f-afe9-15ec91da2cf6'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('cc5bbeba-3226-3af9-8a57-dac0f194d722'): ['127.0.0.3'], UUID('1d30df25-f9a0-335b-b92a-8a2c1c6c3cea'): ['127.0.0.1'], UUID('ba395bad-a383-3b5b-85c7-dbb78c052131'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('662499a0-7a54-3cda-9a80-637581df85f7'): ['127.0.0.3'], UUID('aea261f3-e231-3761-a7b3-d3a3c8b4e859'): ['127.0.0.2'], UUID('1d7d438b-225b-3d92-939e-d95ca030daf4'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('1f9ae93c-96d3-3852-9c6b-6f9e9bd89ae9'): ['127.0.0.3'], UUID('e4d6cb2b-a700-3226-a5c3-3c6b43a73e4e'): ['127.0.0.1'], UUID('78842ada-fefd-389e-82a1-eec9ec58505f'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('21d7c188-8884-3743-ad11-2452311ad73d'): ['127.0.0.1'], UUID('d0fa0794-e686-328b-9979-854ab414ec24'): ['127.0.0.3'], UUID('da011bf9-d865-3014-a489-c4e7e7720b3a'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('89b8caa8-488c-3e00-91ca-8699d4ad7622'): ['127.0.0.2'], UUID('82f01f7f-ed7c-33e5-bbf5-a1f4054aa1d0'): ['127.0.0.3'], UUID('84370337-8e6f-3dda-b75a-330512937cd4'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('8eb695ff-5048-300e-9d39-d30d80648631'): ['127.0.0.1'], UUID('b65f55a6-ca83-3f21-8414-7900e8f2d10f'): ['127.0.0.3'], UUID('e63f1444-e69a-33e6-8358-4993aa4ae5b0'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('58e2f71b-ad57-33d3-b2c9-814a8b8efead'): ['127.0.0.1'], UUID('9acbdce9-07b9-3cd0-a2f6-eddbd335387c'): ['127.0.0.2'], UUID('af4b964d-5eca-3a7f-8ed8-4eeb54e8c1ef'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('9a4f2e33-8e5a-3042-b79b-dfc2f5731a96'): ['127.0.0.2'], UUID('21b4e891-8bc0-38bf-a9d5-a18a06504863'): ['127.0.0.3'], UUID('6372cd47-1386-3248-8288-5fc01da79334'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('fea7ce53-b6ec-34a3-a85e-61fd186cf552'): ['127.0.0.2', '127.0.0.1'], UUID('e0bb9545-52e2-30aa-8c4c-478f90d266bc'): ['127.0.0.3']}
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 192.321s

FAILED (errors=1)
```

**Comment by tgrabiec** [comment]
@penberg I think the original report was a duplicate of #1088. The fact that you are getting bad_alloc may mean we have a regression. Can you check with commit a919113fdb60af1e4ca66042789d70941be65d55? @avikivity, did you ran the dtest after your fix?

**Comment by penberg** [comment]
@tgrabiec I was testing with ``master`` so it has the commit you mention. Or did you mean _without_ the commit?

**Comment by tgrabiec** [comment]
5 kwi 2016 7:14 PM "Pekka Enberg" <notifications@github.com> napisał(a):
>
> @tgrabiec I was testing with master so it has the commit you mention. Or
did you mean without the commit?

I meant to test at exactly that commit to see if it is a regression
introduced after avi's fix or something setup specific

**Comment by penberg** [comment]
@tgrabiec Aye, aye. I'll re-test.

**Comment by penberg** [comment]
@tgrabiec, the test fails exactly the same way with a919113fdb60af1e4ca66042789d70941be65d55:

```
[penberg@nero scylla-dtest]$ nosetests -v concurrent_schema_changes_test.py:TestConcurrentSchemaChanges.create_lots_of_tables_concurrently_test
create_lots_of_tables_concurrently_test (concurrent_schema_changes_test.TestConcurrentSchemaChanges) ... ERROR

======================================================================
ERROR: create_lots_of_tables_concurrently_test (concurrent_schema_changes_test.TestConcurrentSchemaChanges)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/penberg/scylla-dtest/concurrent_schema_changes_test.py", line 146, in create_lots_of_tables_concurrently_test
    results = execute_concurrent(session, cmds, raise_on_first_error=True, concurrency=200)
  File "cassandra/concurrent.py", line 88, in cassandra.concurrent.execute_concurrent (cassandra/concurrent.c:1404)
    return executor.execute(concurrency, raise_on_first_error)
  File "cassandra/concurrent.py", line 185, in cassandra.concurrent.ConcurrentExecutorListResults.execute (cassandra/concurrent.c:4718)
    return super(ConcurrentExecutorListResults, self).execute(concurrency, fail_fast)
  File "cassandra/concurrent.py", line 111, in cassandra.concurrent._ConcurrentExecutor.execute (cassandra/concurrent.c:2059)
    return self._results()
  File "cassandra/concurrent.py", line 199, in cassandra.concurrent.ConcurrentExecutorListResults._results (cassandra/concurrent.c:5459)
    with self._condition:
  File "cassandra/concurrent.py", line 203, in cassandra.concurrent.ConcurrentExecutorListResults._results (cassandra/concurrent.c:5397)
    self._raise(self._exception)
  File "cassandra/concurrent.py", line 151, in cassandra.concurrent._ConcurrentExecutor._raise (cassandra/concurrent.c:3422)
    raise exc
ServerError: <ErrorMessage code=0000 [Server error] message="std::bad_alloc">
-------------------- >> begin captured logging << --------------------
dtest: DEBUG: cluster ccm directory: /home/penberg/.dtest/dtest-wGnxfT
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('ce6b045d-9024-3cc1-b4cd-4429973475aa'): ['127.0.0.3'], UUID('1468039c-0279-3349-9b3d-7e03740967ec'): ['127.0.0.1'], UUID('90e9e8d9-376f-3e13-b222-41935c89ef3b'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.3 is reporting a schema disagreement: {UUID('e3115f8c-fed5-35a9-be7f-e4f43daa5636'): ['127.0.0.3'], UUID('688c345e-0cae-383f-91af-bfae7c565f99'): ['127.0.0.1'], UUID('e03f9fff-c92f-3bce-9572-f185c4a60095'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('f760a254-a29d-3880-bd3a-96dd726ff4d0'): ['127.0.0.3'], UUID('b671cd9c-47d3-3591-a196-8cc335e91dfa'): ['127.0.0.2'], UUID('40d09e02-e3e6-3274-ae7b-796131d14c0a'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('dbfab8e3-99b8-3a09-bc0a-0e5a5ffed83c'): ['127.0.0.1'], UUID('5c17261a-fb9d-3585-ab84-47bc331028e8'): ['127.0.0.2'], UUID('48b64f4d-b5e4-3a83-b52a-df640a046a4c'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('b26315bd-dd5d-333e-a112-363bdfb10b0b'): ['127.0.0.3'], UUID('f38d4c48-3986-3a38-b4df-765f4605da82'): ['127.0.0.1'], UUID('90dbe496-0290-372e-9cd5-51495fb89545'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('a1efeffb-9643-38fd-b1f1-96bbdc06ba9c'): ['127.0.0.3'], UUID('658fc0ad-fbc2-3f28-905c-cf2b20fe65c6'): ['127.0.0.2'], UUID('39826f6c-8697-37df-b237-a05996d426a0'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('d0f646a8-5fd1-3d0c-82c5-f7891e4059fc'): ['127.0.0.3'], UUID('5207d61a-7ad3-3c53-9bb5-5ebb81cb20a9'): ['127.0.0.1'], UUID('9c57a7c1-9053-3e27-a837-3c1662b83d41'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('c97bc61d-7662-3e1b-8c69-df794d413a06'): ['127.0.0.2'], UUID('56dd0ea3-ca7d-3e0e-a4b1-f26d61a7c7dd'): ['127.0.0.1'], UUID('6d2eea6f-3873-397f-8c28-d9e04cf9ee6c'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('4fd6d8ec-93ce-3a68-b307-e2b8e670aae1'): ['127.0.0.1'], UUID('c843dafe-1007-3602-ab18-4baf56a3cd98'): ['127.0.0.3'], UUID('b33a37fe-cd19-3a65-b6e8-7de8e53297b6'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.3 is reporting a schema disagreement: {UUID('fa2a96f4-0a81-3635-8996-fe29af4f3852'): ['127.0.0.2'], UUID('da795d2e-4c97-348f-b02d-6fca8f3ef64a'): ['127.0.0.1'], UUID('9b20c75d-d4f5-3878-a6fb-724eabee21be'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('1e36fe6d-915a-3200-bab4-7c47956d8385'): ['127.0.0.3', '127.0.0.2'], UUID('2f362f60-deb8-3d65-a98b-c9e7329c8f09'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('db31a8a1-1f4e-3765-b57b-74d2f3a43f26'): ['127.0.0.3'], UUID('4b149a93-6ff7-35b6-b46e-92df69f5234f'): ['127.0.0.2'], UUID('7fdb034d-3b5e-32f7-8dc3-e46712fd10f6'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.3 is reporting a schema disagreement: {UUID('4fc26b53-57ca-384f-83cb-f6e85882885b'): ['127.0.0.2'], UUID('b5f97b59-e9f3-305d-82a8-017d4461ddf5'): ['127.0.0.1'], UUID('4b149a93-6ff7-35b6-b46e-92df69f5234f'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('4680628a-4cbb-37b3-bec6-5575a8604246'): ['127.0.0.1'], UUID('b5f97b59-e9f3-305d-82a8-017d4461ddf5'): ['127.0.0.3'], UUID('4b696581-6147-3382-9596-f63aa64552f8'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('7841cae8-a47b-3233-ac30-43a3bfe64b3c'): ['127.0.0.2', '127.0.0.1'], UUID('b5f97b59-e9f3-305d-82a8-017d4461ddf5'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.3 is reporting a schema disagreement: {UUID('7841cae8-a47b-3233-ac30-43a3bfe64b3c'): ['127.0.0.2', '127.0.0.1'], UUID('b35a8d0c-7f73-3b57-9173-5702e9b500c5'): ['127.0.0.3']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('c55b54b9-b029-3ebe-9a32-1adb072c3b21'): ['127.0.0.3', '127.0.0.1'], UUID('c0899367-a842-3e58-aa6a-d6953928f269'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.2 is reporting a schema disagreement: {UUID('c55b54b9-b029-3ebe-9a32-1adb072c3b21'): ['127.0.0.1'], UUID('37009637-c158-39cb-871c-b19a74e3e598'): ['127.0.0.3', '127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('2370a1c5-46f3-3bb2-9275-667c85402078'): ['127.0.0.3'], UUID('37009637-c158-39cb-871c-b19a74e3e598'): ['127.0.0.1'], UUID('b4ee9db8-645e-3135-92f8-d64add98adf6'): ['127.0.0.2']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('06d1c4b3-bcc6-36a3-835b-3d4e14ca868e'): ['127.0.0.2'], UUID('05fab9bf-7166-3b4a-be81-f9d1bf2fade0'): ['127.0.0.3'], UUID('37009637-c158-39cb-871c-b19a74e3e598'): ['127.0.0.1']}
cassandra.cluster: WARNING: Node 127.0.0.1 is reporting a schema disagreement: {UUID('05fab9bf-7166-3b4a-be81-f9d1bf2fade0'): ['127.0.0.3'], UUID('37009637-c158-39cb-871c-b19a74e3e598'): ['127.0.0.1'], UUID('5a611f9f-b1d3-3b15-a5c1-dc63795e534d'): ['127.0.0.2']}
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 255.552s

FAILED (errors=1)
```

**Comment by tzach** [comment]
@slivne ping

**Comment by slivne** [comment]
using 1.7.rc1 ea4a2dad96451470865c2390aa2c3882e28f9642

this test is failing now with SMP=1 as well ...

the failure is different if the wait time is changed and is non consistent:
- ``` ServerError: <ErrorMessage code=0000 [Server error] message="std::bad_alloc">```
- ``` AlreadyExists: Table 'lots_o_tables.t_60' already exists```
- ```ServerError: <ErrorMessage code=0000 [Server error] message="Can't find a column family with UUID 75bfcfc0-08e4-11e7-a889-000000000000">```

export SCYLLA_EXT_OPTS="--smp 2 --memory 1G" it fails as well - tried once got bad_alloc

**Comment by tgrabiec** [comment]
@slivne This test is not expected to pass, we don't allow concurrent schema changes yet (#420,  #1207). 

`std::bad_alloc` may be an indication of some unrelated issue though.

**Comment by slivne** [comment]
@tgrabiec I know we don't allow updates to the same table in parallel.
This test creates multiple different tables in parallel. 
What is limiting the ability to create multiple different tables in parallel ?

**Comment by tgrabiec** [comment]
@slivne I couldn't think of a reason why that would fail, so maybe there is indeed an issue.

**Comment by slivne** [comment]
the only thing I can think off is if the driver does retransmission of
create tables and that may cause one of the issues above.

I'll try to see if we can disable retransmission and then see if the test
passes

**Comment by tgrabiec** [comment]
Fixed by 350d98d4e1e5dcf257ad73e2d46eda60e3d6a17a
